File: archived/DisplayOrientation/js/js/scenario1-orientation.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";

    // Convenient aliases for WinRT types.
    var DisplayInformation = Windows.Graphics.Display.DisplayInformation;
    var DisplayOrientations = Windows.Graphics.Display.DisplayOrientations;

    // Elements in the HTML page.
    var nativeOrientationSpan;
    var currentOrientationSpan;
    var autoRotationPreferencesSpan;
    var preferLandscape;
    var preferPortrait;
    var preferLandscapeFlipped;
    var preferPortraitFlipped;

    // Per-page state.
    var displayInfo;

    var page = WinJS.UI.Pages.define("/html/scenario1-orientation.html", {
        ready: function (element, options) {
            nativeOrientationSpan = document.getElementById("nativeOrientationSpan");
            currentOrientationSpan = document.getElementById("currentOrientationSpan");
            autoRotationPreferencesSpan = document.getElementById("autoRotationPreferencesSpan");
            preferLandscape = document.getElementById("preferLandscape");
            preferPortrait = document.getElementById("preferPortrait");
            preferLandscapeFlipped = document.getElementById("preferLandscapeFlipped");
            preferPortraitFlipped = document.getElementById("preferPortraitFlipped");

            document.getElementById("applyRotationLock").addEventListener("click", applyRotationLock, false);

            displayInfo = DisplayInformation.getForCurrentView();
            displayInfo.addEventListener("orientationchanged", updateContent, false);

            updateContent();
        },

        unload: function () {
            displayInfo.removeEventListener("orientationchanged", updateContent);
        }
    });

    function orientationsToString(orientations) {
        var flags = [];
        if (orientations & DisplayOrientations.landscape)
        {
            flags.push("Landscape");
        }
        if (orientations & DisplayOrientations.portrait)
        {
            flags.push("Portrait");
        }
        if (orientations & DisplayOrientations.landscapeFlipped)
        {
            flags.push("LandscapeFlipped");
        }
        if (orientations & DisplayOrientations.portraitFlipped)
        {
            flags.push("PortraitFlipped");
        }
        return flags.join(", ") || "None";
    }

    function updateContent()
    {
        nativeOrientationSpan.innerText = orientationsToString(displayInfo.nativeOrientation);
        currentOrientationSpan.innerText = orientationsToString(displayInfo.currentOrientation);
        autoRotationPreferencesSpan.innerText = orientationsToString(DisplayInformation.autoRotationPreferences);

    }

    function applyRotationLock()
    {
        var orientations = DisplayOrientations.none;
        if (preferLandscape.checked)
        {
            orientations |= DisplayOrientations.landscape;
        }
        if (preferPortrait.checked)
        {
            orientations |= DisplayOrientations.portrait;
        }
        if (preferLandscapeFlipped.checked)
        {
            orientations |= DisplayOrientations.landscapeFlipped;
        }
        if (preferPortraitFlipped.checked)
        {
            orientations |= DisplayOrientations.portraitFlipped;
        }
        DisplayInformation.autoRotationPreferences = orientations;
        updateContent();

    }
})();

File: Samples/DisplayOrientation/cppwinrt/Scenario1_Orientation.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario1_Orientation.h"
#include "Scenario1_Orientation.g.cpp"

using namespace winrt;
using namespace winrt::Windows::Foundation;
using namespace winrt::Windows::Graphics::Display;
using namespace winrt::Windows::UI::Xaml;

namespace winrt::SDKTemplate::implementation
{
    Scenario1_Orientation::Scenario1_Orientation()
    {
        InitializeComponent();
    }

    void Scenario1_Orientation::OnNavigatedTo(Windows::UI::Xaml::Navigation::NavigationEventArgs const&)
    {
        displayInfo = DisplayInformation::GetForCurrentView();
        orientationChangedToken = displayInfo.OrientationChanged({ get_weak(), &Scenario1_Orientation::OnOrientationChanged });
        UpdateContent();
    }

    void Scenario1_Orientation::OnNavigatedFrom(Windows::UI::Xaml::Navigation::NavigationEventArgs const&)
    {
        displayInfo.OrientationChanged(orientationChangedToken);
    }

    void Scenario1_Orientation::OnOrientationChanged(DisplayInformation const&, IInspectable const&)
    {
        UpdateContent();
    }

    hstring Scenario1_Orientation::OrientationsToString(DisplayOrientations orientations)
    {
        std::wstring result = L"None";

        if ((orientations & DisplayOrientations::Landscape) == DisplayOrientations::Landscape)
        {
            result += L", Landscape";
        }
        if ((orientations & DisplayOrientations::Portrait) == DisplayOrientations::Portrait)
        {
            result += L", Portrait";
        }
        if ((orientations & DisplayOrientations::LandscapeFlipped) == DisplayOrientations::LandscapeFlipped)
        {
            result += L", LandscapeFlipped";
        }
        if ((orientations & DisplayOrientations::PortraitFlipped) == DisplayOrientations::PortraitFlipped)
        {
            result += L", PortraitFlipped";
        }
        if (result != L"None")
        {
            result = result.substr(6);
        }
        return hstring{ result };
    }

    void Scenario1_Orientation::UpdateContent()
    {
        NativeOrientationTextBlock().Text(OrientationsToString(displayInfo.NativeOrientation()));
        CurrentOrientationTextBlock().Text(OrientationsToString(displayInfo.CurrentOrientation()));
        AutoRotationPreferencesTextBlock().Text(OrientationsToString(DisplayInformation::AutoRotationPreferences()));
    }

    void Scenario1_Orientation::ApplyRotationLock_Click(IInspectable const&, RoutedEventArgs const&)
    {
        DisplayOrientations orientations = DisplayOrientations::None;

        if (PreferLandscape().IsChecked().Value())
        {
            orientations = orientations | DisplayOrientations::Landscape;
        }
        if (PreferPortrait().IsChecked().Value())
        {
            orientations = orientations | DisplayOrientations::Portrait;
        }
        if (PreferLandscapeFlipped().IsChecked().Value())
        {
            orientations = orientations | DisplayOrientations::LandscapeFlipped;
        }
        if (PreferPortraitFlipped().IsChecked().Value())
        {
            orientations = orientations | DisplayOrientations::PortraitFlipped;
        }
        DisplayInformation::AutoRotationPreferences(orientations);
        UpdateContent();
    }
}
